R/data_type.R

```diff
@@ -234,3 +239,35 @@ split_raw_data <- function(df,
     x[, selected_columns, with = FALSE]
   }, df_list)
 }
+
+#' @keywords internal
+#'
+unify_combination_data <- function(df, cl, drug_ids) {
+  cotrt_data <- split(df, df[[cl]])
+  data.table::rbindlist(lapply(cotrt_data, function(x) {
+    duplicated_full_idx <- which(x[[drug_ids[["drug_name"]]]] %chin% x[[drug_ids[["drug_name2"]]]] &
+                                   x[[drug_ids[["drug_name2"]]]] %chin% x[[drug_ids[["drug_name"]]]])
+    if (length(duplicated_full_idx)) {
+      duplicated_data <- x[duplicated_full_idx, ]
+      drug_data <- duplicated_data[, drug_ids[c("drug_name", "drug_name2")], with = FALSE]
+      unique_rows <- drug_data[!duplicated(t(apply(drug_data, 1, sort))), ]
+      idx_duplicated <- duplicated_full_idx[which(is.na(matches(do.call(paste, unique_rows),
+                                                                do.call(paste, drug_data))$x))]
+      
+      primary_drug_idfs <- drug_ids[c("drug", "drug_name", "drug_moa", "concentration")]
+      secondary_drug_idfs <- drug_ids[c("drug2", "drug_name2", "drug_moa2", "concentration2")]
+      
+      x[idx_duplicated,
```

Comment:
@bczech : I think (but not sure) that there is an issue because other variables (like drug3, or any additional rowData) are ignored in the swap. We should only when `durg` and `drug2` (and `drug_name`, `drug_moa`) are *the only* swapped values. 
We should also try to account for `drug3`, but this is more complicated and could be done later.

Reply:
Hi Marc, with Drug3 the procedure might be more tricky. Let's do it in a separate task later.

Reply:
https://jira.gene.com/jira/browse/GDR-2246